eland/operations.py

```diff
@@ -530,14 +573,67 @@ def _unpack_metric_aggs(
                         else:
                             agg_value = field.np_dtype.type(agg_value)
 
-                values.append(agg_value)
+                if not percentile_values:
+                    values.append(agg_value)
 
             # If numeric_only is True and We only have a NaN type field then we check for empty.
             if values:
                 results[field.column] = values if len(values) > 1 else values[0]
+            # This only runs when df.quantile() or series.quantile() is called
+            if percentile_values and not is_dataframe_agg:
+                results[f"{field.column}"] = percentile_values
 
         return results
 
+    def quantile(
+        self,
+        query_compiler: "QueryCompiler",
+        pd_aggs: List[str],
+        quantiles: Union[float, List[float]],
+        is_dataframe: bool = True,
+        numeric_only: bool = True,
+    ) -> Union[pd.DataFrame, pd.Series]:
+        # To verify if quantile range falls between 0 to 1
+        def verify_quantile_range(quantile: Any) -> float:
+            if isinstance(quantile, (int, float, str)):
+                quantile = float(quantile)
+                if quantile > 1 or quantile < 0:
+                    raise ValueError(
+                        f"quantile should be in range of 0 and 1, given {quantile}"
+                    )
+            else:
+                raise TypeError("quantile should be of type int or float or str")
+            # quantile * 100 = percentile
+            return quantile * 100
+
+        percentiles = list(
```

Comment:
Prefer list-comprehensions instead of `list(map(...))`

```python
if not isinstance(quantiles, (list, tuple)):
    quantiles = (quantiles,)
percentiles = [quantile_to_percentile(x) for x in quantiles]
```